lib/Mojo/Redis/Connection.pm

```diff
@@ -46,8 +46,12 @@ sub write_q {
 
 sub _encode {
   my $self     = shift;
+  my $protocol = $self->protocol;
+
+  return $protocol->encoding($self->encoding)->encode(\@_) if $protocol->can('encoding');
```

Comment:
I think this is too fragile, it should check ->isa('Mojo::Protocol::Redis') specifically because the encode method has different behavior.